### ISSUE TITLE
Failed to create detector when loading tensorrt yolox model by C SDK.

### ISSUE THREAD

**Issue by codehasbug** [open-mmlab/mmdeploy#145]
Step:  
1. Training yolox model base on the "yolox/yolox_m_8x8_300e_coco.py" configuration.  
2. Convert to trt model by MMDeploy. Config:  
```
_base_ = ['../_base_/base_tensorrt_static-300x300.py']

onnx_config = dict(input_shape=(640, 640))

backend_config = dict(
    common_config=dict(max_workspace_size=1 << 30),
    model_inputs=[
        dict(
            input_shapes=dict(
                input=dict(
                    min_shape=[1, 3, 640, 640],
                    opt_shape=[1, 3, 640, 640],
                    max_shape=[1, 3, 640, 640])))
    ])
```
3.  According to [run mmdeploy sdk demo](https://mmdeploy.readthedocs.io/en/latest/get_started.html#run-mmdeploy-sdk-demo)
 run mmdeploy sdk demo.
Log:
```
[2022-02-10 03:05:07.150] [mmdeploy] [info] [model.cpp:97] Register 'DirectoryModel'
[2022-02-10 03:05:07.290] [mmdeploy] [info] [common.h:15] config: {
  "context": {
    "device": "<any>",
    "stream": "<any>"
  },
  "input": [
    "image"
  ],
  "name": "mmdetection",
  "output": [
    "det"
  ],
  "params": {
    "model": "/headless/data/sources/python/deploy/yolox_m_with_carparts"
  },
  "type": "Inference"
}
[2022-02-10 03:05:07.291] [mmdeploy] [info] [model.cpp:45] DirectoryModel successfully load sdk model /headless/data/sources/python/deploy/yolox_m_with_carparts
[2022-02-10 03:05:07.291] [mmdeploy] [info] [common.h:15] config: {
  "context": {
    "device": "<any>",
    "model": "<any>",
    "stream": "<any>"
  },
  "input": [
    "img"
  ],
  "module": "Transform",
  "name": "Preprocess",
  "output": [
    "prep_output"
  ],
  "transforms": [
    {
      "type": "LoadImageFromFile"
    },
    {
      "keep_ratio": false,
      "size": [
        640,
        640
      ],
      "type": "Resize"
    },
    {
      "pad_to_square": true,
      "pad_val": {
        "img": [
          114.0,
          114.0,
          114.0
        ]
      },
      "size_divisor": 1,
      "type": "Pad"
    },
    {
      "keys": [
        "img"
      ],
      "type": "ImageToTensor"
    },
    {
      "keys": [
        "img"
      ],
      "meta_keys": [
        "flip_direction",
        "img_shape",
        "img_norm_cfg",
        "ori_shape",
        "scale_factor",
        "filename",
        "valid_ratio",
        "flip",
        "ori_filename",
        "pad_shape"
      ],
      "type": "Collect"
    }
  ],
  "type": "Task"
}
[2022-02-10 03:05:07.292] [mmdeploy] [info] [common.h:15] config: {
  "context": {
    "device": "<any>",
    "model": "<any>",
    "stream": "<any>"
  },
  "input": [
    "img"
  ],
  "module": "Transform",
  "name": "Preprocess",
  "output": [
    "prep_output"
  ],
  "transforms": [
    {
      "type": "LoadImageFromFile"
    },
    {
      "keep_ratio": false,
      "size": [
        640,
        640
      ],
      "type": "Resize"
    },
    {
      "pad_to_square": true,
      "pad_val": {
        "img": [
          114.0,
          114.0,
          114.0
        ]
      },
      "size_divisor": 1,
      "type": "Pad"
    },
    {
      "keys": [
        "img"
      ],
      "type": "ImageToTensor"
    },
    {
      "keys": [
        "img"
      ],
      "meta_keys": [
        "flip_direction",
        "img_shape",
        "img_norm_cfg",
        "ori_shape",
        "scale_factor",
        "filename",
        "valid_ratio",
        "flip",
        "ori_filename",
        "pad_shape"
      ],
      "type": "Collect"
    }
  ],
  "type": "Task"
}
[2022-02-10 03:05:07.293] [mmdeploy] [error] [detector.cpp:58] exception caught: invalid argument (1) @ /headless/data/sources/python/mmdeploy/csrc/core/value.h:543
Failed to create detector, code: 6

Process finished with exit code 255
```
4. Tracking error
Debugging the demo found the problem occur in `${MMDEPLOY_DIR}/csrc/graph/common.h:21`
`auto inst = creator->Create(config);`  
After the function,  A exception has been throw by `/headless/data/sources/python/mmdeploy/csrc/core/value.h:543`
`config` is the third config of log:
```
[2022-02-10 03:05:07.292] [mmdeploy] [info] [common.h:15] config: {
  "context": {
    "device": "<any>",
    "model": "<any>",
    "stream": "<any>"
  },
  "input": [
    "img"
  ],
  "module": "Transform",
  "name": "Preprocess",
  "output": [
    "prep_output"
  ],
  "transforms": [
    {
      "type": "LoadImageFromFile"
    },
    {
      "keep_ratio": false,
      "size": [
        640,
        640
      ],
      "type": "Resize"
    },
    {
      "pad_to_square": true,
      "pad_val": {
        "img": [
          114.0,
          114.0,
          114.0
        ]
      },
      "size_divisor": 1,
      "type": "Pad"
    },
    {
      "keys": [
        "img"
      ],
      "type": "ImageToTensor"
    },
    {
      "keys": [
        "img"
      ],
      "meta_keys": [
        "flip_direction",
        "img_shape",
        "img_norm_cfg",
        "ori_shape",
        "scale_factor",
        "filename",
        "valid_ratio",
        "flip",
        "ori_filename",
        "pad_shape"
      ],
      "type": "Collect"
    }
  ],
  "type": "Task"
}
```
I thought the problem may be related to the third configuration,  Can you help me?  think you.

**Comment by codehasbug** [comment]
Error occur in ${MMDEPLOY_DIR}/csrc/preprocess/transform/pad.cpp:88   
```
Pad::Pad(const Value& args, int version) : Transform(args) {
  auto impl_creator = Registry<PadImpl>::Get().GetCreator(specified_platform_, version);
  if (nullptr == impl_creator) {
    ERROR("'Pad' is not supported on '{}' platform", specified_platform_);
    throw std::domain_error("'Pad' is not supported on specified platform");
  }
  impl_ = impl_creator->Create(args); // In this function
}
```
args:
```
{
  "context": {
    "device": "<any>",
    "model": "<any>",
    "stream": "<any>"
  },
  "pad_to_square": true,
  "pad_val": {
    "img": [
      114.0,
      114.0,
      114.0
    ]
  },
  "size_divisor": 1,
  "type": "Pad"
}

```

**Comment by codehasbug** [comment]
The configuration does not match the `pad_val` data type in the data structure.
```
struct pad_arg_t {
    std::array<int, 2> size;
    int size_divisor;
    float pad_val;
    bool pad_to_square;
    std::string padding_mode;
  };
```

**Comment by codehasbug** [comment]
Finally, In `${work_dirs}/pipeline.json` ,  modify `pad_val` option, RESOLVED!
```
......
"pad_val": 114.0,
......
```
But I think it would be better if this problem could be handled automatically during the model transition.

**Comment by lvhan028** [comment]
Thanks for reporting the issue. We'll fix it asap

**Comment by Bo396543018** [comment]
@codehasbug 请问下你的yolox Demo跑通了是么~ 我按照步骤跑到最后一步报了一个错误

```
[ERROR][2022-02-21 09:58:21.068][resize.cu:1202] CUDA error: no kernel image is available for execution on the device
[2022-02-21 09:58:21.068] [mmdeploy] [error] error: unknown (6) @ /mnt/Deploy/mmdeploy/csrc/preprocess/cuda/resize_impl.cpp:88
[2022-02-21 09:58:21.068] [mmdeploy] [error] unhandled exception: unknown (6) @ /mnt/Deploy/mmdeploy/csrc/preprocess/cuda/resize_impl.cpp:88
[2022-02-21 09:58:21.068] [mmdeploy] [error] exception caught: unknown (6) @ /mnt/Deploy/mmdeploy/csrc/experimental/module_adapter.h:35
```
看起来是cuda版本不一致问题，但是我编译使用的都是一个版本，不确定哪里出了问题，想请教下大佬有遇到类似问题么0.0 俺的环境如下:
```
2022-02-21 10:00:42,533 - mmdeploy - INFO -                                                                                                                   
                                                                                                                                                              
2022-02-21 10:00:42,533 - mmdeploy - INFO - **********Environmental information**********                                                                     
2022-02-21 10:00:43,978 - mmdeploy - INFO - sys.platform: linux                                                                                               
2022-02-21 10:00:43,979 - mmdeploy - INFO - Python: 3.7.10 | packaged by conda-forge | (default, Feb 19 2021, 16:07:37) [GCC 9.3.0]                           
2022-02-21 10:00:43,979 - mmdeploy - INFO - CUDA available: True                                                                                              
2022-02-21 10:00:43,979 - mmdeploy - INFO - GPU 0,1,2,3,4,5,6,7: GeForce RTX 3090                                                                             
2022-02-21 10:00:43,979 - mmdeploy - INFO - CUDA_HOME: /usr/local/cuda                                                                                        
2022-02-21 10:00:43,979 - mmdeploy - INFO - NVCC: Build cuda_11.2.r11.2/compiler.29373293_0                                                                   
2022-02-21 10:00:43,979 - mmdeploy - INFO - GCC: gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0                                                                      
2022-02-21 10:00:43,979 - mmdeploy - INFO - PyTorch: 1.8.0                   
2022-02-21 10:00:43,979 - mmdeploy - INFO - PyTorch compiling details: PyTorch built with:                                                                    
  - GCC 7.3                                                                                                                                                   
  - C++ Version: 201402                                                                                                                                       
  - Intel(R) oneAPI Math Kernel Library Version 2021.2-Product Build 20210312 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v1.7.0 (Git Hash 7aed236906b1f7a05c0917e5257a1af05e9ff683)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 11.1
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=comput$_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_8$;-gencode;arch=compute_37,code=compute_37
  - CuDNN 8.0.5
  - Magma 2.5.2
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.1, CUDNN_VERSION=8.0.5, CXX_COMPILER=/opt/rh/devtoolset-7/root/usr/bin/c++, CXX_FLAGS=
-Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -$2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-co$pare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing $Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-co$or=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, LAPAC$_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.8.0, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GL$G=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON,

2022-02-21 10:00:43,979 - mmdeploy - INFO - TorchVision: 0.9.0
2022-02-21 10:00:43,979 - mmdeploy - INFO - OpenCV: 4.5.3
2022-02-21 10:00:43,979 - mmdeploy - INFO - MMCV: 1.4.0
2022-02-21 10:00:43,979 - mmdeploy - INFO - MMCV Compiler: GCC 7.3
2022-02-21 10:00:43,979 - mmdeploy - INFO - MMCV CUDA Compiler: 11.1
2022-02-21 10:00:43,979 - mmdeploy - INFO - MMDeployment: 0.2.0+59470fe
2022-02-21 10:00:43,979 - mmdeploy - INFO -

2022-02-21 10:00:43,979 - mmdeploy - INFO - **********Backend information**********
2022-02-21 10:00:44,209 - mmdeploy - INFO - onnxruntime: 1.8.1 ops_is_avaliable : True
2022-02-21 10:00:44,209 - mmdeploy - INFO - tensorrt: 7.2.3.4 ops_is_avaliable : True
2022-02-21 10:00:44,210 - mmdeploy - INFO - ncnn: None ops_is_avaliable : False
2022-02-21 10:00:44,210 - mmdeploy - INFO - pplnn_is_avaliable: False
2022-02-21 10:00:44,210 - mmdeploy - INFO - openvino_is_avaliable: False
2022-02-21 10:00:44,210 - mmdeploy - INFO -

2022-02-21 10:00:44,210 - mmdeploy - INFO - **********Codebase information**********
2022-02-21 10:00:44,210 - mmdeploy - INFO - mmcls: None
2022-02-21 10:00:44,211 - mmdeploy - INFO - mmdet: 2.21.0
2022-02-21 10:00:44,211 - mmdeploy - INFO - mmedit: None
2022-02-21 10:00:44,211 - mmdeploy - INFO - mmocr: None
2022-02-21 10:00:44,211 - mmdeploy - INFO - mmseg: None
```

**Comment by codehasbug** [comment]
@Bo396543018 Sorry, I haven't seen the issue before. It looks like an environmental issue, try to update Nvidia driver and build all components from source.

**Comment by Bo396543018** [comment]
@codehasbug  好的，感谢答复，我在排查下原因

**Comment by lvhan028** [comment]
> @codehasbug 好的，感谢答复，我在排查下原因

"CUDA error: no kernel image is available for execution on the device" is usually caused by the mismatched CUDA compute capability and NVIDIA GPU.

**Comment by codehasbug** [comment]
The problem has been solved by this pr #164 .
